src/Rematch2.jl

```diff
@@ -19,6 +19,10 @@ macro _const(x)
     end
 end
 
+is_expr(e, head) = e isa Expr && e.head == head
+is_expr(e, head, n) = is_expr(e, head) && length(e.args) == n
+is_case(e) = is_expr(e, :call, 3) && e.args[1] == :(=>)
```

Comment:
to match the base Julia implementation, we probably want to wrap all these `e` in `@nospecialize(e)`